src/twitter/Timeline.js

```diff
@@ -215,115 +264,141 @@ class Timeline {
       else{
         this._hashtags = queue;
       }
-
+      return true;
     }
     else {
       log.warn("Tried to remove %s but it doesn't exist", id);
+      return false;
     }
   }
 
-  _process_timeline () {
-    if (this._timelines.length === 0) {
-      return;
+  is_room_excluded (rooms) {
+    if (!this.config.timelines.poll_if_empty) {
+      const difference = new Set([...rooms].filter(x => !this._empty_rooms.has(x)));
+      if (difference.size === 0) {
+        return true;
+      }
     }
+    return false;
+  }
 
-    var tline = this._timelines[this._t];
-    var req = {
-      user_id: tline.twitter_id,
-      count: TIMELINE_TWEET_FETCH_COUNT,
-      exclude_replies: tline.exclude_replies,
-      tweet_mode: "extended" // https://github.com/Half-Shot/matrix-appservice-twitter/issues/31
-    };
-
-    if(this._newtags.has(req.user_id)) {
-      req.count = 1;
-      this._newtags.delete(req.user_id);
+  is_feed_exceeding_user_limit (tweets, timeline = true) {
+    let max = (timeline ? (TIMELINE_POLL_INTERVAL*this._t) : (HASHTAG_POLL_INTERVAL*this._h))/60000;
+    max = Math.max(max * NEW_PROFILE_THRESHOLD_MIN, 1)
+    if ((timeline ? this.config.timelines : this.config.hashtags).single_account_fallback === true) {
+      const user_ids = new Set(tweets.map((tweet) => {tweet.id_str})).size;
+      log.verbose(`is_feed_exceeding_user_limit: ${user_ids} > ${max}`);
```

Comment:
It might be more useful to log verbosely if the feed _is_ exceeding the user limit. Stating the number of ids and the max doesn't seem useful.